back/src/Application/Buying/Handler/DeleteBuyingCommandHandler.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Application\Buying\Handler;
+
+use App\Application\Buying\Command\DeleteBuyingCommand;
+use App\Domain\Buying\Repository\BuyingRepositoryInterface;
+use App\Domain\Common\Exception\ForbiddenException;
+
+class DeleteBuyingCommandHandler
+{
+    public function __construct(
+        private readonly BuyingRepositoryInterface $buyingRepository,
+    ) {
+    }
+
+    public function __invoke(DeleteBuyingCommand $command): void
+    {
+        $buying = $this->buyingRepository->getOneByUid($command->buyingId);
+        $user = $command->user;
+
+        if ($buying->getUser() !== $user) {
+            throw new ForbiddenException(sprintf('User %s cannot delete buying %s', $user->getEmail(), $buying->getUid()->toRfc4122()));
+        }
+
+        $this->buyingRepository->delete($buying);
```

Comment:
```suggestion
        return $this->buyingRepository->delete($buying);
```
Le delete du repo va renvoyer un boolean, on va le modifier plus bas :)